tests/integration/test_record.py

```diff
@@ -28,304 +28,263 @@
+            '$ref': 'http://localhost:5000/api/literature/111',
+        },
+    }
+
+    deleted_record = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'control_number': 222,
+        'self': {
+            '$ref': 'http://localhost:5000/api/literature/222',
+        },
+    }
 
     with app.app_context():
-        json_record_merged = hep.do(create_record(snippet_merged))
-        json_record_merged['$schema'] = 'http://localhost:5000/schemas/records/hep.json'
-
-        json_record_deleted = hep.do(create_record(snippet_deleted))
-        json_record_deleted['$schema'] = 'http://localhost:5000/schemas/records/hep.json'
-
         with db.session.begin_nested():
-            merged_id = record_upsert(json_record_merged).id
-            deleted_id = record_upsert(json_record_deleted).id
-
+            merged_uuid = record_upsert(merged_record).id
+            deleted_uuid = record_upsert(deleted_record).id
         db.session.commit()
 
     yield
 
     with app.app_context():
-        _delete_merged_records_from_everywhere(
-            'lit',
-            111,
-            222,
-            merged_id,
-            deleted_id
-        )
-
+        _delete_merged_records('lit', 111, 222, merged_uuid, deleted_uuid)
 
-def _delete_record_from_everywhere(pid_type, record_control_number):
-    record = get_db_record(pid_type, record_control_number)
 
-    record._delete(force=True)
-
-    pid = PersistentIdentifier.get(pid_type, record_control_number)
-    PersistentIdentifier.delete(pid)
-
-    object_uuid = pid.object_uuid
-    PersistentIdentifier.query.filter(
-        object_uuid == PersistentIdentifier.object_uuid).delete()
+@pytest.fixture(scope='function')
+def records_to_be_merged(app):
+    merged_record = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'control_number': 111,
+        'self': {
+            '$ref': 'http://localhost:5000/api/literature/111',
+        },
+    }
+
+    deleted_record = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'control_number': 222,
+        'self': {
+            '$ref': 'http://localhost:5000/api/literature/222',
+        },
+    }
+
+    pointing_record = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'accelerator_experiments': [
+            {
+                'record': {
+                    '$ref': 'http://localhost:5000/api/literature/222',
+                },
+            },
+        ],
+        'control_number': 333,
+        'self': {
+            '$ref': 'http://localhost:5000/api/literature/333',
+        },
+    }
 
-    db.session.commit()
+    with app.app_context():
+        with db.session.begin_nested():
+            merged_uuid = record_upsert(merged_record).id
+            deleted_uuid = record_upsert(deleted_record).id
+            record_upsert(pointing_record)
+        db.session.commit()
+    es.indices.refresh('records-hep')
```

Comment:
This is why we shouldn't search in ES to do DB modifications: there's no guarantee that they are in sync.